### PR TITLE
fix(fieldset): support spacing between fieldset in form - FE-3485

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
fixes: #3528
fix margin spacing between fieldset components

### Current behaviour
there was a fix but it was overrided by different style
### Checklist

- [x] Commits follow our style guide
- [ ] <del> Screenshots are included in the PR if useful
- [ ] <del> All themes are supported if required
- [ ] <del> Unit tests added or updated if required
- [ ] <del> Cypress automation tests added or updated if required
- [ ] <del> Storybook added or updated if required
- [ ] <del> Typescript `d.ts` file added or updated if required
- [ ] <del> Carbon implementation and Design System documentation are congruent

### Additional context
More ampersands added because it is related to ampersands that are used in `fieldset.style.js`

### Testing instructions
https://codesandbox.io/s/boring-mayer-db3vb

### CODE REVIEW

File: src/__experimental__/components/fieldset/fieldset.stories.mdx [hunk 1]
```diff
@@ -56,7 +57,46 @@ import Fieldset from "carbon-react/lib/__experimental__/components/fieldset";
...
+          <Textbox label="Fieldset 2 Field 1" labelInline />
+          <Textbox label="Fieldset 2 Field 2" labelInline />
+        </Fieldset>
+        <Textbox label="Separate Field" labelInline />
+      </Form>
+  </Story>
+</Preview>
+
+### `fieldSpacing`
+You can manipulate the spacings beetwen `Fieldset` components by passing `fieldsetSpacing` prop into `Form` component
```
Comment (edleeks87) [inline src/__experimental__/components/fieldset/fieldset.stories.mdx:80]:
```suggestion
You can manipulate the spacings beetwen `Fieldset` components by passing `fieldSpacing` prop into `Form` component
```

File: src/__experimental__/components/fieldset/fieldset.stories.mdx [hunk 2]
```diff
@@ -56,7 +57,46 @@ import Fieldset from "carbon-react/lib/__experimental__/components/fieldset";
...
+        <Fieldset>
+          <Textbox label="Fieldset 2 Field 1" labelInline />
+          <Textbox label="Fieldset 2 Field 2" labelInline />
+        </Fieldset>
+        <Textbox label="Separate Field" labelInline />
+      </Form>
+  </Story>
+</Preview>
+
+### Validations
```
Comment (edleeks87) [inline src/__experimental__/components/fieldset/fieldset.stories.mdx:99]:
```suggestion
## Validations
```

File: src/components/form/form.style.js [hunk 3]
```diff
@@ -15,6 +16,11 @@ export const StyledForm = styled.form`
       theme.spacing * fieldSpacing}px;
   }
 
+  & ${FieldsetStyle} {
```
Comment (edleeks87) [inline src/components/form/form.style.js:19]: I don't think you need to define this block again, see above suggested change

File: src/components/form/form.style.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ import { StyledFieldset } from "../../__internal__/fieldset/fieldset.style";
 import StyledButton from "../button/button.style";
 import baseTheme from "../../style/themes/base";
 import OptionsHelper from "../../utils/helpers/options-helper";
+import { FieldsetStyle } from "../../__experimental__/components/fieldset/fieldset.style";
 
 export const StyledForm = styled.form`
   & ${StyledFormField}, ${StyledFieldset}, > ${StyledButton} {
```
Comment (edleeks87) [inline src/components/form/form.style.js:13]:
```suggestion
  & ${StyledFormField}, ${StyledFieldset}, ${FieldsetStyle}, > ${StyledButton} {
```

File: src/components/form/form.spec.js [hunk 5]
```diff
@@ -48,6 +50,41 @@ describe("Form", () => {
...
+    wrapper = mount(<StyledForm />);
+
+    it("as default", () => {
+      assertStyleMatch(
+        {
+          marginTop: "0",
+          marginBottom: "24px",
+        },
+        wrapper,
+        {
```
Comment (edleeks87) [inline src/components/form/form.spec.js:63]: This can be 
```
 {  modifier: `${FieldsetStyle}`, }
```

File: src/components/form/form.spec.js [hunk 6]
```diff
@@ -48,6 +50,41 @@ describe("Form", () => {
...
+    it("as custom value", () => {
+      wrapper = mount(<StyledForm fieldSpacing={2} />);
+      assertStyleMatch(
+        {
+          marginTop: "0",
+          marginBottom: "16px",
+        },
+        wrapper,
+        {
+          modifier: css`
```
Comment (edleeks87) [inline src/components/form/form.spec.js:80]: This can be 
```
 {  modifier: `${FieldsetStyle}`, }
```

File: src/components/form/form.spec.js [hunk 7]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import { mount, shallow } from "enzyme";
 import { act } from "react-dom/test-utils";
 
+import { css } from "styled-components";
```
Comment (edleeks87) [inline src/components/form/form.spec.js:5]: I don't think you need to import the `css` module for the tests to work

**Review by DipperTheDan — COMMENTED** [review]
Looks good. As discussed, could you add the `Testing Instructions` to the PR description please? Thanks 😄

**Review by edleeks87 — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**